src/libsyntax/ext/build.rs

```diff
@@ -1004,7 +1004,7 @@ impl<'a> AstBuilder for ExtCtxt<'a> {
         // FIXME: Would be nice if our generated code didn't violate
         // Rust coding conventions
         P(ast::Item {
-            ident: name,
+            ident: name.dummy_spanned(),
```

Comment:
Similarly here. I feel like `dummy_spanned` is basically always bad, since it implies that we can't just cite an error at `item.name.span`, but rather we have to maybe fallback to `item.span`.

This perhaps suggests that rather than having us do `item.name.span`, we should add `item.best_span` or something, which is the span to use in an error (versus being the span of the entire item in the source).